deepchecks/tabular/metric_utils/scorers.py

```diff
@@ -222,8 +225,56 @@ def run_on_pred(self, y_true, y_pred=None, y_proba=None):
             return self.scorer._score_func(y_true, pred_to_use, **self.scorer._kwargs) * self.scorer._sign
         raise errors.DeepchecksValueError('Only supports sklearn scorers')
 
+    def _wrap_classification_model(self, model, label_col: pd.Series):
+        """Convert labels to 0/1 if model is a binary classifier, update classes_ property and pad probas."""
+
+        class MyModelWrapper:
+            """Convert labels to 0/1 if model is a binary classifier, update classes_ property and pad probas."""
+
+            def __init__(self, user_model, possible_classes):
+                self.user_model = user_model
+                self.possible_classes = possible_classes
+                if hasattr(model, 'classes_') and len(model.classes_) > 0 \
+                        and len(model.classes_) != len(possible_classes):
+                    self.indexes_to_pad_around = [possible_classes.index(x) for x in list(model.classes_)]
+                else:
+                    self.indexes_to_pad_around = None
+
+            def predict(self, data: pd.DataFrame) -> np.ndarray:
+                if len(self.possible_classes) == 2:
+                    transfer_func = np.vectorize(lambda x: 0 if x == self.possible_classes[0] else 1)
+                    return transfer_func(np.asarray(self.user_model.predict(data)))
+                else:
+                    return self.user_model.predict(data)
+
+            def predict_proba(self, data: pd.DataFrame) -> np.ndarray:
+                probabilities_per_class = self.user_model.predict_proba(data)
+                if self.indexes_to_pad_around is not None:
+                    padded_probabilities_per_class = np.zeros((len(data), len(self.possible_classes)))
+                    for idx, i in enumerate(self.indexes_to_pad_around):
+                        padded_probabilities_per_class[:, i] = probabilities_per_class[:, idx]
+                    return padded_probabilities_per_class
+                elif probabilities_per_class.shape[1] != len(self.possible_classes):
+                    raise errors.ModelValidationError(
+                        f'Model probabilities per class has {probabilities_per_class.shape[1]} '
```

Comment:
Not clear from here that the problem is inconsistent lengths

Reply:
it is not